GPUQREngine/Source/BucketList/BucketList.cpp

```diff
@@ -133,3 +133,5 @@ void BucketList::Initialize()
         }
     }
 }
+extern template class BucketList<int64_t>;
+extern template class BucketList<int32_t>;
```

Comment:
These `extern template` declarations at the end of a .cpp file can probably be removed.
Think of them like a forward declaration of a class that is defined in a different compilation unit.

What might make sense to do here is to explicitly instantiate the `template class` with those two types:
``` suggestion
template class BucketList<int64_t>;
template class BucketList<int32_t>;
```

The same for the other .cpp files.